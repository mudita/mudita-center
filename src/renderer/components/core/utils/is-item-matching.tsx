/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/LICENSE.md
 */

export type IsItemMatching = (item: any, searchString: string) => boolean
export const isItemValueMatching: IsItemMatching = (
  itemValue: string,
  search
) => itemValue.toLowerCase().includes(search.toLowerCase())
