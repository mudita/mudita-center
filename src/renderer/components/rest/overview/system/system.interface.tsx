/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/LICENSE.md
 */

export interface SystemProps {
  osVersion: string
  lastUpdate: string
  onUpdateCheck?: () => void
  onUpdate?: () => void
  onDownload?: () => void
  updateAvailable?: boolean
  updateDownloaded?: boolean
}
