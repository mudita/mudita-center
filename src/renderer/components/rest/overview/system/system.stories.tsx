/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/LICENSE.md
 */

import { storiesOf } from "@storybook/react"
import React, { useEffect, useState } from "react"
import System from "Renderer/components/rest/overview/system/system.component"
import getFakeAdapters from "App/tests/get-fake-adapters"
import { action } from "@storybook/addon-actions"
import { css } from "styled-components"
import StoryContainer from "Renderer/components/storybook/story-container.component"
import Story from "Renderer/components/storybook/story.component"

const fakeSystemInfo = getFakeAdapters().purePhone
const updated = new Date(fakeSystemInfo.getOsUpdateDate()).toLocaleDateString(
  "en-US"
)

const storyStyle = css`
  > * {
    width: 59rem;
  }
`

storiesOf("Views|Overview/System", module).add("System", () => {
  const [osVersion, setOsVersion] = useState("")
  useEffect(() => {
    const fetch = async () => {
      const { data = "" } = await fakeSystemInfo.getOsVersion()
      setOsVersion(data)
    }
    fetch()
  }, [])

  return (
    <StoryContainer column>
      <Story title="Not updated yet" customStyle={storyStyle}>
        <System
          osVersion={osVersion}
          onUpdateCheck={action("checked for update")}
          lastUpdate={"2020-01-14T11:31:08.244Z"}
        />
      </Story>
      <Story title="Updated some time ago" customStyle={storyStyle}>
        <System
          osVersion={osVersion}
          lastUpdate={updated}
          onUpdateCheck={action("checked for update")}
        />
      </Story>
      <Story title="New update available" customStyle={storyStyle}>
        <System
          osVersion={osVersion}
          lastUpdate={updated}
          onUpdate={action("update requested")}
          updateAvailable
        />
      </Story>
      <Story title="New (first) update available" customStyle={storyStyle}>
        <System
          osVersion={osVersion}
          onUpdate={action("update requested")}
          updateAvailable
          lastUpdate={"2020-01-14T11:31:08.244Z"}
        />
      </Story>
      <Story title="Updated just now (custom text)" customStyle={storyStyle}>
        <System
          osVersion={osVersion}
          lastUpdate={"just now"}
          onUpdateCheck={action("checked for update")}
        />
      </Story>
    </StoryContainer>
  )
})
