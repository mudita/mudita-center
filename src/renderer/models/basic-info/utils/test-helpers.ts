/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/LICENSE.md
 */

import { IpcRequest } from "Common/requests/ipc-request.enum"
import DeviceResponse, {
  DeviceResponseStatus,
} from "Backend/adapters/device-response.interface"

const makeSuccessDeviceResponse = async (
  data: any
): Promise<DeviceResponse<any>> => {
  return {
    data,
    status: DeviceResponseStatus.Ok,
  }
}

export const commonCalls = {
  [IpcRequest.GetDeviceInfo]: makeSuccessDeviceResponse({
    name: "Ziemniaczek",
    modelName: "U12300000",
    modelNumber: "A1239999",
    serilaNumber: "a-b-3d",
    osVersion: "0.123v",
    osUpdateDate: "12-12-2003",
  }),
  [IpcRequest.GetNetworkInfo]: makeSuccessDeviceResponse({
    simCards: [
      {
        active: true,
        network: "Y-Mobile",
        networkLevel: 0.5,
        number: 12345678,
        slot: 1,
      },
      {
        active: false,
        network: "X-Mobile",
        networkLevel: 0.69,
        number: 7001234523,
        slot: 2,
      },
    ],
  }),
  [IpcRequest.GetStorageInfo]: makeSuccessDeviceResponse({
    capacity: 9001,
    available: 99999999999999,
    categories: [
      { label: "music", filesCount: 1233333, size: 999999999 },
      { label: "storage", filesCount: 100000, size: 999999999 },
    ],
  }),
  [IpcRequest.GetBatteryInfo]: makeSuccessDeviceResponse({
    level: 9001,
    charging: false,
    maximumCapacity: 99999,
  }),
  [IpcRequest.GetBackupsInfo]: makeSuccessDeviceResponse({
    backups: [
      {
        createdAt: "20-11-15T07:35:01.562Z20",
        size: 99999,
      },
      {
        createdAt: "20-01-30T07:35:01.562Z20",
        size: 1234567,
      },
    ],
  }),
}
