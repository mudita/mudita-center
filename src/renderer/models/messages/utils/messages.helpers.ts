/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/LICENSE.md
 */

import { Contact } from "App/contacts/store/contacts.type"
import { Topic } from "Renderer/models/messages/messages.interface"

export type ContactsCollection = Record<string, Contact>
export type Getter = (
  id: string,
  collection: ContactsCollection
) => Contact | undefined

export const getContactsAsMap = (contacts: Contact[]) => {
  return contacts.reduce((acc: ContactsCollection, item: Contact) => {
    return {
      ...acc,
      [item.id]: item,
    }
  }, {})
}

export const getContactDetails = (
  id: string,
  collection: ContactsCollection
): Contact | undefined => {
  if (id in collection) {
    return collection[id]
  }

  return undefined
}

export const expandTopic = (
  topic: Topic,
  collection: ContactsCollection,
  getter: Getter
) => {
  const { messages } = topic

  return {
    ...topic,
    messages: messages.map((msg) => {
      const author = getter(msg.author.id, collection)

      if (author) {
        return {
          ...msg,
          author,
        }
      }

      return msg
    }),
  }
}

export const createFullMessagesCollection = (state: {
  messages: { topics: Topic[] }
  contacts: { db: ContactsCollection }
}): Topic[] => {
  const {
    messages: { topics },
    contacts: { db: baseContacts },
  } = state

  return topics.map(
    (topic: Topic): Topic => {
      const { id, phoneNumber } = topic.caller
      const contact: Contact = baseContacts[id]

      if (contact) {
        const { firstName, lastName } = contact
        return {
          ...expandTopic(topic, baseContacts, getContactDetails),
          caller: { id, phoneNumber, firstName, lastName },
        }
      }

      return topic
    }
  )
}
