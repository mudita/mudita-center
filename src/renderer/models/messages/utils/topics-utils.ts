/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/LICENSE.md
 */

import {
  MessagesState as MessagesProps,
  Topic,
  VisibilityFilter,
} from "Renderer/models/messages/messages.interface"

export const searchTopics = (
  topics: MessagesProps["topics"],
  searchValue: MessagesProps["searchValue"]
) => {
  if (searchValue.length) {
    return topics?.filter(({ caller, messages }) => {
      const search = searchValue.toLowerCase()
      const matchesForename = caller.firstName?.toLowerCase().includes(search)
      const matchesSurname = caller.lastName?.toLowerCase().includes(search)
      const matchesPhone = caller.phoneNumber?.includes(search)
      const matchesMessages = messages.some(({ content }) =>
        content.toLowerCase().includes(search)
      )
      return (
        matchesForename || matchesSurname || matchesPhone || matchesMessages
      )
    })
  } else {
    return topics
  }
}
export const filterTopics = (
  topics: MessagesProps["topics"],
  visibilityFilter: MessagesProps["visibilityFilter"]
) =>
  topics?.filter(({ unread }) =>
    visibilityFilter === VisibilityFilter.Unread ? unread : true
  )

export const sortTopics = (topics: MessagesProps["topics"]) => {
  const lastMessageDate = ({ messages }: Topic) => {
    return messages[messages.length - 1].date
  }
  return topics?.sort((a, b) => {
    const x = lastMessageDate(a)
    const y = lastMessageDate(b)
    return x > y ? -1 : x < y ? 1 : 0
  })
}
