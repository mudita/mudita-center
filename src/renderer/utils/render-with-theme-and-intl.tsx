/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/LICENSE.md
 */

import {
  Queries,
  render,
  RenderOptions,
  RenderResult,
} from "@testing-library/react"
import React from "react"
import { IntlProvider } from "react-intl"
import { defaultLanguage } from "App/translations.config.json"
import localeEn from "Renderer/locales/default/en-US.json"
import { ThemeProvider } from "styled-components"
import theme from "../styles/theming/theme"
import extractLanguageKeys from "Renderer/utils/extract-test-locale"

const testLocale = extractLanguageKeys(localeEn)

export function renderWithThemeAndIntl(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult
export function renderWithThemeAndIntl<Q extends Queries>(
  ui: React.ReactElement,
  options: RenderOptions<Q>
): RenderResult
export function renderWithThemeAndIntl<Q extends Queries>(
  ui: React.ReactElement,
  options?: RenderOptions<Q>
): RenderResult {
  return render<Q>(
    <ThemeProvider theme={theme}>
      <IntlProvider
        defaultLocale={defaultLanguage}
        locale={defaultLanguage}
        messages={process.env.NODE_ENV === "test" ? testLocale : localeEn}
      >
        {ui}
      </IntlProvider>
    </ThemeProvider>,
    // @ts-ignore
    options
  )
}
