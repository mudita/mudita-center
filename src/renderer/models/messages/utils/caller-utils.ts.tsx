import { Caller } from "Renderer/models/calls/calls.interface"
import { removeDecoratorsFromPhoneNumber } from "Renderer/models/utils/remove-decorators-from-phone-number"

export const isCallerMatchingPhoneNumber = (
  caller: Caller,
  phoneNumber: string
): boolean => {
  if (phoneNumber === "") {
    return false
  }

  return (
    removeDecoratorsFromPhoneNumber(caller.phoneNumber) ===
    removeDecoratorsFromPhoneNumber(phoneNumber)
  )
}
