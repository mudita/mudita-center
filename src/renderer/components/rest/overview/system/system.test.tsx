/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/LICENSE.md
 */

import "@testing-library/jest-dom/extend-expect"
import React from "react"
import { renderWithThemeAndIntl } from "Renderer/utils/render-with-theme-and-intl"
import { SystemProps } from "Renderer/components/rest/overview/system/system.interface"
import getFakeAdapters from "App/tests/get-fake-adapters"
import { intl } from "Renderer/utils/intl"
import System from "Renderer/components/rest/overview/system/system.component"
import { fireEvent } from "@testing-library/dom"
import { waitFor } from "@testing-library/react"

const fakeSystemInfo = getFakeAdapters().purePhone
const fakeLastUpdate = "2020-01-14T11:31:08.244Z"
let fakeOsVersion: string

beforeEach(async () => {
  const { data } = await fakeSystemInfo.getOsVersion()
  fakeOsVersion = data ?? ""
})

const renderSystem = ({
  osVersion = fakeOsVersion,
  lastUpdate = fakeLastUpdate,
  ...props
}: Partial<SystemProps> = {}) => {
  return renderWithThemeAndIntl(
    <System osVersion={osVersion} lastUpdate={lastUpdate} {...props} />
  )
}

test("matches snapshot", () => {
  const { container } = renderSystem()
  expect(container).toMatchSnapshot()
})

test("renders os version properly", () => {
  const { getByText } = renderSystem()
  expect(getByText(fakeOsVersion)).toBeInTheDocument()
  expect(
    getByText(intl.formatMessage({ id: "view.name.overview.system.version" }))
  ).toBeInTheDocument()
})

test("renders last update info properly", () => {
  const { getByText } = renderSystem()
  expect(
    getByText(
      intl.formatMessage(
        { id: "view.name.overview.system.lastUpdate" },
        { date: fakeLastUpdate }
      )
    )
  ).toBeInTheDocument()
})

test("renders available update info properly", () => {
  const { getByText } = renderSystem({ updateAvailable: true })
  expect(
    getByText(
      intl.formatMessage({ id: "view.name.overview.system.updateAvailable" })
    )
  ).toBeInTheDocument()
})

test("renders 'check for updates' button properly", () => {
  const { queryByRole } = renderSystem()
  expect(queryByRole("button")).toHaveTextContent(
    intl.formatMessage({ id: "view.name.overview.system.checkForUpdates" })
  )
})

test("renders 'update now' button properly", () => {
  const { queryByRole } = renderSystem({ updateAvailable: true })
  expect(queryByRole("button")).toHaveTextContent(
    intl.formatMessage({ id: "view.name.overview.system.downloadAction" })
  )
})

test("checks for update after button click", async () => {
  const onUpdateCheck = jest.fn()

  const { getByRole } = renderSystem({
    onUpdateCheck,
  })

  fireEvent.click(getByRole("button"))

  await waitFor(() => {
    expect(onUpdateCheck).toHaveBeenCalled()
  })
})

test("triggers download after button click", async () => {
  const onDownload = jest.fn()

  const { getByRole } = renderSystem({
    updateAvailable: true,
    onDownload,
  })

  fireEvent.click(getByRole("button"))

  await waitFor(() => {
    expect(onDownload).toHaveBeenCalled()
  })
})

test("triggers update after button click", async () => {
  const onUpdate = jest.fn()

  const { getByRole } = renderSystem({
    updateAvailable: true,
    updateDownloaded: true,
    onUpdate,
  })

  fireEvent.click(getByRole("button"))

  await waitFor(() => {
    expect(onUpdate).toHaveBeenCalled()
  })
})
