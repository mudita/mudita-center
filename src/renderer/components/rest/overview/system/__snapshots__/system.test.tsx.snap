// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
.c3 {
  margin: 0;
  font-size: 2.4rem;
  line-height: 0.8;
  font-weight: 700;
  -webkit-letter-spacing: -0.06rem;
  -moz-letter-spacing: -0.06rem;
  -ms-letter-spacing: -0.06rem;
  letter-spacing: -0.06rem;
}

.c4 {
  margin: 0;
  font-size: 1.2rem;
  line-height: 1.2;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
}

.c5 {
  margin: 0;
  font-size: 1.2rem;
  line-height: 1.2;
  color: #6a6a6a;
  -webkit-letter-spacing: 0.04rem;
  -moz-letter-spacing: 0.04rem;
  -ms-letter-spacing: 0.04rem;
  letter-spacing: 0.04rem;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 2.8rem;
  width: 2.8rem;
  position: relative;
  font-style: normal;
}

.c12 svg {
  width: 100%;
  height: 100%;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  -webkit-transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  cursor: pointer;
  outline: none;
  box-sizing: border-box;
  width: 17rem;
  height: 4rem;
  color: #ffffff;
  border-radius: 0.5rem;
  background: #3b3f42;
  border: 0.1rem solid #3b3f42;
}

.c9:hover {
  background: #232426;
}

.c9 g {
  fill: #ffffff;
}

.c13 {
  margin: 0 0.8rem 0 0;
}

.c14 {
  white-space: nowrap;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  position: relative;
}

.c10 {
  --radius: 0.5rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  z-index: 1;
  pointer-events: all;
  z-index: 3;
}

.c10:disabled {
  pointer-events: none;
}

.c10:hover {
  z-index: 2;
}

.c10:not(:first-of-type) {
  border-radius: 0;
  margin-left: -0.1rem;
}

.c10:first-of-type:not(:last-of-type) {
  border-radius: var(--radius) 0 0 var(--radius);
}

.c10:last-of-type:not(:first-of-type) {
  border-radius: 0 var(--radius) var(--radius) 0;
}

.c0 {
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-template-areas: "Text Buttons";
  grid-template-columns: auto minmax(18rem,1fr);
  padding: 0 4.8rem;
  box-sizing: border-box;
  border-radius: 0.5rem;
  background-color: #ffffff;
}

.c1 {
  grid-area: Text;
}

.c8 {
  grid-area: Buttons;
  justify-self: end;
  min-width: 18rem;
}

.c11 {
  padding: 0 1.6rem;
  width: 50%;
}

.c11 svg {
  height: initial;
  width: initial;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.c2 span {
  margin-left: 0.8rem;
  font-weight: 300;
  text-transform: none;
}

.c6 {
  margin-top: 1.2rem;
  -webkit-letter-spacing: 0.04rem;
  -moz-letter-spacing: 0.04rem;
  -ms-letter-spacing: 0.04rem;
  letter-spacing: 0.04rem;
  color: #6a6a6a;
}

<div>
  <section
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          release-0.55.1
        </h2>
        <span
          class="c4"
        >
          [value] view.name.overview.system.version
        </span>
      </div>
      <p
        class="c5 c6"
      >
        [value] view.name.overview.system.lastUpdate
      </p>
    </div>
    <section
      class="c7 c8"
    >
      <button
        class="c9 c10 c11"
        size="2"
        type="button"
      >
        <i
          class="c12 c13"
          data-testid="icon-Reload"
          height="2.8"
          width="2.8"
        >
          <test-file-stub
            image="test-file-stub"
          />
        </i>
        <p
          class="c4 c14"
        >
          [value] view.name.overview.system.checkForUpdates
        </p>
      </button>
    </section>
  </section>
</div>
`;
